CMakeLists.txt

```diff
@@ -1,6 +1,9 @@
-cmake_minimum_required(VERSION 3.12)
+cmake_minimum_required(VERSION 3.12...3.16)
```

Comment:
Whats the logic behind adding range ?

Reply:
To ensure users that this project has been tested on all these versions. However it is not true, since neither I have tested not I have updated travis to test with all these cmake versions. What I can do is, I can update travis to add additional 4 cases to test all cmake versions or if you think it is not necessary at this point, I will remove this change.

Reply:
Too much trouble I would say. CMake is doing minor releases only, they won't break much. We can directly increase min version to 3.17 (latest release) if you are fine with it.

Reply:
I am fine with minimum version to v3.17. Do you want me to make that as part of this branch or will you make that change?

Reply:
Sure, just change this to min version of 3.17

Reply:
Done.